### PR TITLE
Release/4.7.0

### LINKED ISSUE #157: Release version 4.7.0
This issue is for tracking the 4.7.0 release. Target release date: **June 25 2024**

## Pre-release steps

- [x] Resolve conflits & merge #139      
- [x] Resolve conflits & merge #147      
- [x] Merge #105   
- [x] Merge #127  
- [x] Merge #145 
   
## Release steps

- [x] Branch: Starting from `trunk`, cut a release branch named `release/4.7.0`
- [x] Version bump: Bump the version number in `woocommerce-square.php`, `package.json`, `package-lock.json` and `readme.txt` to 4.7.0. In `woocommerce-square.php` update both the plugin "Version:" property and the plugin `WC_SQUARE_VERSION` constant
- [x] Update `@since`: Find all new `@since x.x.x` / `@since n.e.x.t` lines and update those with the new version number, 4.7.0
- [x] Generate changelog: Locally, generate the changelog by running the following Woorelease command: `php woorelease.phar cl:generate --product_version=4.7.0 https://github.com/woocommerce/woocommerce-square/tree/trunk`. Ensure the changelog updates look correct
- [x] Update changelog: Take the changelog generated from the previous step and update the release PR with those changes, both in the `changelog.txt` file and `readme.txt` file
- [x] Open PR: Open a new PR against `trunk` from the release branch, `release/4.7.0`
- [x] Run tests: On this release PR, add the label `needs: qit default tests` and `needs: e2e testing`. This will trigger the default QIT tests (activation, API and E2E) and our E2E tests. Ensure these tests pass as well as all other applicable tests
- [x] Merge: Merge the release PR into `trunk`
- [x] Prepare ZIP: Prepare the zip by running Woorelease locally on the `trunk` branch: `php woorelease.phar build https://github.com/woocommerce/woocommerce-square/tree/trunk`. Make sure you are running the proper versions of `node`, `npm` and `composer` to ensure the built files are consistent across releases. Versions can be found in [`package.json`](https://github.com/woocommerce/woocommerce-square/blob/trunk/package.json)
- [x] Compare: Download the latest [released zip](https://github.com/woocommerce/woocommerce-square/releases) and then run a diff compare against the generated zip, ensuring all expected changes are there (and no unexpected changes)
- [x] Test ZIP: Take the ZIP generated from the previous step and test this locally, ensuring version numbers look correct, the changelog looks correct and the plugin installs and activates properly
- [x] Attach ZIP: Attach the ZIP to this issue and ping the Automattic team in the `#a8c-10up-woo-support` channel to complete the release

### PR DESCRIPTION
**Pull request by dkotter**
### Changes proposed in this Pull Request:

Version bump and prep for the 4.7.0 release

Closes #157 

### Steps to test the changes in this Pull Request:

Verify all file changes look accurate

### CODE REVIEW

**Comment by dkotter** [comment]
E2E tests were failing and I eventually cancelled them, as they just kept retrying and failing. It seems like basically all tests are failing so I'm assuming it's some sort of configuration issue but we'll need to resolve this before proceeding

**Review by vikrampm1 — APPROVED** [review]

**Comment by Sidsector9** [comment]
59/61 tests are passing locally on this branch. More tests are failing on pipeline.

cc: @faisal-alvi 

localhost_9323_

**Comment by iamdharmesh** [comment]
@dkotter @vikrampm1 The E2E tests check is successful and we are good to proceed with the next steps here.
Note: I noticed some sync-related tests are flaky.

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,12 @@
 *** WooCommerce Square Changelog ***
 
+= 4.7.0 - 2024-06-25 =
```
Comment (dkotter) [inline changelog.txt]:
```suggestion
= 4.7.0 - 2024-06-27 =
```

File: readme.txt [hunk 2]
```diff
@@ -72,6 +72,13 @@ If you get stuck, you can ask for help in the [Plugin Forum](https://wordpress.o
 
 == Changelog ==
 
+= 4.7.0 - 2024-06-25 =
```
Comment (dkotter) [inline readme.txt]:
```suggestion
= 4.7.0 - 2024-06-27 =
```

**Pull request merged**